File: src/lecture03/java/Main.java
package lecture03.java;

public class Main {

    public static void printAgeIfPerson(Object obj) {
        if (obj instanceof Person) {
            Person person = (Person) obj;
            System.out.println(person.getAge());
        }
    }

    public static String personInformation(Person person) {
        return String.format("사람의 이름은 %s이고 나이는 %s세 입니다.",
                person.getName(),
                person.getAge());
    }

    public static void main(String[] args) {
        int number1 = 4;
        long number2 = number1; // 더 큰 타입으로 암시적으로 형변환
        System.out.println(number1 + number2);

        Person person = new Person("who-hoo", 100);
        printAgeIfPerson(person);

        System.out.println(personInformation(person));
    }
}

File: src/lecture03/kotlin/Main.kt
package lecture03.kotlin

import lecture03.java.Person

fun printAgeIfPerson(obj: Any?) {
    val person = obj as? Person
    println(person?.age)
}

fun personInformation(person: Person) : String {
    return "사람의 이름은 ${person.name}이고 나이는 ${person.age}세 입니다."
}

fun main() {
    val number1 = 4
    val number2: Long = number1.toLong() // 명시적 형변환 필요
    println(number1 + number2)

    val person = Person("who-hoo", 100)
    printAgeIfPerson(null)
    printAgeIfPerson(person)

    println(personInformation(person))

    val multilineStr = """
        여러 줄로
        작성하는
        문자열!!!
    """.trimIndent()
    println(multilineStr)

    val str = "abcdefg"
    println(str[0])
}
